pkg/config/config.go

```diff
@@ -214,25 +214,16 @@ func New(filepath string) (*Config, error) {
 		return nil, fmt.Errorf("replication slot name is not specified")
 	}
 
-	connCfg, err := pgx.ParseEnvLibpq()
+	connConfig, err := pgx.ParseConfig(cfg.Postgres.UrlOrDsnConnstring)
 	if err != nil {
-		return nil, fmt.Errorf("could not parse lib pq env variabels: %v", err)
+		return nil, fmt.Errorf("could not parse lib pq env variabels or supplied conn string: %v", err)
 	}
+	cfg.Postgres.config = connConfig
 
 	if cfg.InactivityFlushTimeout.Seconds() == 0 {
 		cfg.InactivityFlushTimeout = defaultInactivityMergeTimeout
 	}
 
-	cfg.Postgres.ConnConfig = cfg.Postgres.ConnConfig.Merge(connCfg)
-
-	if cfg.Postgres.Port == 0 {
-		cfg.Postgres.Port = defaultPostgresPort
-	}
-
-	if cfg.Postgres.Host == "" {
-		cfg.Postgres.Host = defaultPostgresHost
-	}
-
```

Comment:
`pgx/v4` favors taking pg connection as input via a connection string in either url or dsn format.